src/stash/src/lib.rs

```diff
@@ -75,6 +75,9 @@ pub trait Stash: std::fmt::Debug + Send {
         K: Data,
         V: Data;
 
+    /// Returns the names of the collections in the stash.
+    async fn collections(&mut self) -> Result<BTreeSet<String>, StashError>;
```

Comment:
This might be too big of a refactor for this PR, but if we could somehow create a method like
`async fn collections(&mut self) -> Result<BTreeSet<Box<dyn TypedCollectionTrait>>, StashError>`
then we might be able to get rid of the hard-coded collections in `stash-debug` and use that method instead. 

I briefly tried and failed to see if it would even compile, so it's probably not worth it. Just thought I'd mention it in case it sparked any ideas.

Reply:
I'm not sure this is possible? If it is that'd be neat!